### PR TITLE
Address audit report comments

### LINKED ISSUE #202: Fix Audit security comments
in app/src/apdu_sign.c : 486, buffer’s content as to be zeroized (i.e.,cleared). For this, only the buffer size was reset to zero. In doing so, the data persists in memory, with the risk of being read by another channel.

- [x] Reset the buffer and not just its size. 
- [x] Initialize `pressed_right` in ui_stream.c and ui_stream_nbgl.c
- [x] Fix TODO/NOTE and other comments - check app/src/ui_stream_nbgl.c: 352-354 and app/src/globals.h: 122
- [x] Remove redundant check for global.step in app/src/apdu_sign.c : 486
- [x] <=0 check is not correct for unsigned int. Fix at app/src/parser/operation_parser.c: 418
- [x] fix printf format specifier incompatibility at app/src/apdu_sign.c : 585

### PR DESCRIPTION
**Pull request by ajinkyaraj-23**
fixes #202

### CODE REVIEW

File: app/src/apdu.c [hunk 1]
```diff
@@ -1,7 +1,5 @@
 /* Tezos Ledger application - Some common primitives and some command handlers
 
-   TODO: split this file (apdu primitives and apdu handlers)
```
Comment (spalmer25) [inline app/src/apdu.c:3]: I'm not sure whether we have any trace of these `TODO`s.
Could you open an issue to make sure we don't forget them?

**Review by spalmer25 — COMMENTED** [review]
LGTM

File: app/src/apdu.c (hunk 1 above)
Comment (ajinkyaraj-23) [reply app/src/apdu.c:3]: Created issues #207 #208 and #209

**Comment by ajinkyaraj-23** [comment]
Rebased on main.

File: app/src/apdu.c (hunk 1 above)
Comment (spalmer25) [reply app/src/apdu.c:3]: What about ?
```
        // TODO: this is handled by change_screen_right except we disable it
        // to use it here. We should make ui_stream fully compatible with
        // exception.h
```
in [app/src/ui_stream_nbgl.c](https://github.com/trilitech/ledger-app-tezos-wallet/pull/203/files#diff-ba33a3ab1ba1a41b1110b6ddf932fc7096b6f05730525c30fb5ffc398fc6dcce)
Is this `TODO` outdated ?

**Review by spalmer25 — COMMENTED** [review]
There is a forgotten `FIXME` in https://github.com/trilitech/ledger-app-tezos-wallet/blob/5b4e04c247bd801c9d78251883e3167051d4922c/app/src/ui_stream.h#L154-L155

File: app/src/apdu.c (hunk 1 above)
Comment (ajinkyaraj-23) [reply app/src/apdu.c:3]: Created #211 .

**Comment by ajinkyaraj-23** [comment]
> There is a forgotten `FIXME` in
> 
> https://github.com/trilitech/ledger-app-tezos-wallet/blob/5b4e04c247bd801c9d78251883e3167051d4922c/app/src/ui_stream.h#L154-L155

Fixed. The issue is same as #211

**Review by spalmer25 — APPROVED** [review]

**Pull request merged**